slatedb/src/manifest/mod.rs

```diff
@@ -70,6 +74,121 @@ impl Manifest {
             compactor_epoch: parent_manifest.compactor_epoch,
         }
     }
+
+    #[allow(unused)]
+    pub(crate) fn projected(source_manifest: &Manifest, range: BytesRange) -> Manifest {
+        let mut projected = source_manifest.clone();
+        let mut sorter_runs_filtered = vec![];
+        for sorter_run in &projected.core.compacted {
+            sorter_runs_filtered.push(SortedRun {
+                id: sorter_run.id,
+                ssts: Self::filter_sst_handles(&sorter_run.ssts, false, &range),
+            });
+        }
+        projected.core.l0 = Self::filter_sst_handles(&projected.core.l0, true, &range).into();
+        projected.core.compacted = sorter_runs_filtered;
+        projected
+    }
+
+    fn filter_sst_handles<'a, T>(
+        handles: T,
+        handles_overlap: bool,
+        projection_range: &BytesRange,
+    ) -> Vec<SsTableHandle>
+    where
+        T: IntoIterator<Item = &'a SsTableHandle>,
+    {
+        let mut iter = handles.into_iter().peekable();
+        let mut filtered_handles = vec![];
+        while let Some(current_handle) = iter.next() {
+            let next_handle = if handles_overlap {
+                None
+            } else {
+                iter.peek().copied()
+            };
+            if let Some(intersection) =
+                current_handle.compacted_intersection(next_handle, projection_range)
+            {
+                filtered_handles.push(current_handle.with_visible_range(intersection));
+            }
+        }
+        filtered_handles
+    }
+
+    #[allow(unused)]
+    pub(crate) fn union(manifests: Vec<Manifest>) -> Manifest {
+        if manifests.len() == 1 {
+            manifests[0].clone()
+        } else {
+            let mut ranges = vec![];
+            for manifest in &manifests {
+                let range = manifest.range();
+                if let Some(range) = range {
+                    ranges.push((manifest, range));
+                } else {
+                    tracing::warn!("Manifest {:?} has no SST files", manifest);
+                }
+            }
+            ranges.sort_by_key(|(_, range)| range.comparable_start_bound().cloned());
+
+            // Ensure manifests are non-overlapping
+            let mut previous_range = None;
+            for (_, range) in ranges.iter() {
+                if let Some(previous_range) = previous_range {
+                    if range.intersect(previous_range).is_some() {
```

Comment:
The assumption here is that the DBs are non overlapping.
Would there be a scenario where they are overlapping? What would we do in that case?